### PR TITLE
endpoint: Add a function to lookup by container ID

### PR DESCRIPTION
**Pull request by michi-covalent**
Also add .idea directory to gitignore.

Signed-off-by: Michi Mutsuzaki <michi@isovalent.com>

### CODE REVIEW

File: pkg/api/v1/endpoint.go [hunk 1]
```diff
@@ -196,3 +196,20 @@ func (es *Endpoints) GetEndpoint(ip net.IP) (endpoint *Endpoint, ok bool) {
...
+func (es *Endpoints) GetEndpointByContainerID(id string) (*Endpoint, bool) {
+	es.mutex.RLock()
+	defer es.mutex.RUnlock()
+	for _, ep := range es.eps {
+		if ep.Deleted != nil {
+			continue
+		}
+		for _, containerID := range ep.ContainerIDs {
+			if id == containerID {
+				return ep, true
```
Comment (Rolinh) [inline pkg/api/v1/endpoint.go]: `es.eps` is a slice of pointer to `Endpoint`s. The caller of that method can thus modify the `Endpoint` and it's probably not something that should be allowed.

Comment (michi-covalent) [reply pkg/api/v1/endpoint.go]: fair point. we do the same thing in GetEndpoint() as well. should we change it to return a copy instead?

Comment (Rolinh) [reply pkg/api/v1/endpoint.go]: I think it should eventually be tackled as part of #76 but for the time being, returning a copy and adding a `FIXME` note seems reasonable to me.

Comment (michi-covalent) [reply pkg/api/v1/endpoint.go]: added another commit to make a copy before returning

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,3 @@
 /hubble
 install/kubernetes/hubble.yaml
+.idea/
```
Comment (glibsm) [inline .gitignore]: This doesn't hurt to add, but this belongs in your global .gitignore, really.

Comment (michi-covalent) [reply .gitignore]: fair enough. reverted.

**Pull request merged**